akka-javasdk/src/main/scala/akka/javasdk/impl/SdkRunner.scala

```diff
@@ -354,6 +358,24 @@ private final class Sdk(
     }
   }
 
+  private def hasCorrectProfile(clz: Class[_]): Boolean = {
+    if (clz.hasAnnotation[Profile]) {
+      val profiles = clz.getAnnotation(classOf[Profile]).value
+      if (profiles.isEmpty || profiles.forall(sdkSettings.profiles.contains)) {
```

Comment:
I'm wondering what would be the best default case here: to require all annotated profiles to be present or to require at least one profile? I see that Spring does it the same way you did it here.. so probably a good default.